### PR TITLE
Unit tests & proposed new format

### PR DESCRIPTION
**Pull request by WalshyDev**
**This is being made from a base of 4.1, make sure to use StreamDeck version 4.1 when testing this branch**

## Changes:
* StreamDeck4J has been renamed to Plugin
  This was done to give a better meaning to what exactly this class is about
* Plugin is now an interface
  Simply put, it's a cleaner class for docs along with more accessiblity to internal methods for other classes and unit tests
* PluginBuilder to just make it a bit easier especially if I expand more on different options for implementing Plugin
* All events now have a `getPlugin()` to return `Plugin`

## Still TODO:
- [ ] Get feedback on the new layout and make any adjustments.
- [ ] Document everything, this includes the events. I'll need to unlombok them.
- [ ] Write unit tests for all public methods
- [ ] Test fully on 4.1 betas
  - [ ] Also just wait for all betas and implement the new features

This will be merged once 4.1 is released as the new stable version.

### CODE REVIEW

**Comment by taydev** [comment]
Yes.

File: src/main/java/com/walshydev/streamdeck4j/Plugin.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+     * @param url The URL to open
+     */
+    void openURL(@Nonnull URL url);
+
+    /**
+     * Sets the title of a specific button on the Stream Deck
+     *
+     * @param context     The unique identifier for the button
+     * @param title       The new title for the button
+     * @param destination The destination for the event (Hardware, Software or Both)
```
Comment (taydev) [inline src/main/java/com/walshydev/streamdeck4j/Plugin.java:62]: What is this, old Destination docs? Please fix, absolutely unusable.

**Review by taydev — DISMISSED** [review]

File: src/main/java/com/walshydev/streamdeck4j/Plugin.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/com/walshydev/streamdeck4j/Plugin.java:62]: Isn't that your job? ;)

Comment (taydev) [reply src/main/java/com/walshydev/streamdeck4j/Plugin.java:62]: I didn't write these files ;)

Comment (WalshyDev) [reply src/main/java/com/walshydev/streamdeck4j/Plugin.java:62]: You wrote the documentation for them, that's enough ;)

**Comment by WalshyDev** [comment]
Update did original comment to include a TODO and notice that I won't merge till 4.1 is stable. 

If anyone can help with docs, please do :)

**Comment by taydev** [comment]
I suppose I'll let your low Codacy score slide.

**Pull request merged**